src/mono/wasm/Wasm.Build.Tests/BuildTestBase.cs

```diff
@@ -839,6 +841,41 @@ protected void AssertBlazorBundle(string config, bool isPublish, bool dotnetWasm
                         dotnetJsPath!,
                         "Expected dotnet.native.js to be same as the runtime pack",
                         same: dotnetWasmFromRuntimePack);
+
+            string bootConfigPath = Path.Combine(binFrameworkDir, "blazor.boot.json");
+            Assert.True(File.Exists(bootConfigPath), $"Expected to find '{bootConfigPath}'");
+
+            using (var bootConfigContent = File.OpenRead(bootConfigPath))
+            {
+                var bootConfig = ParseBootData(bootConfigContent);
+                var dotnetJsEntries = bootConfig.resources.runtime.Keys.Where(k => k.StartsWith("dotnet.") && k.EndsWith(".js")).ToArray();
+
+                void AssertFileExists(string fileName)
+                {
+                    string absolutePath = Path.Combine(binFrameworkDir, fileName);
+                    Assert.True(File.Exists(absolutePath), $"Expected to find '{absolutePath}'");
+                }
+
+                string versionHashRegex = @"\d.0.\d?(-[a-z]+(\.\d\.\d+\.\d)?)?\.([a-zA-Z0-9])+";
```

Comment:
Do we need it to be this specific? The regex that I had pushed should capture the version, and hash without being specific about the format.

Reply:
Sorry, I override it by accident. 

Anyway, since we are checking whether the file is fingerprinted as expected, the more strict the regex is, the better, isn't it?

Reply:
This is the regex I was suggesting: `@"\.(?<version>.+)\.(?<hash>[a-zA-Z0-9]+)\.` .
- it assumes the format for the hash, and combined with the `".js"`, it makes up the end of the string, and thus can be matched easily
- And we know the prefix, like `dotnet.native.`, which allows the remaining part to be extracted as the version.
- We can then check the version if we want, but even if we don't it doesn't depend on the format of the version string. A stricter *version* regex would be more fragile.

Reply:
Ook, I'll revert to your suggestion 👍